src/target/riscv/riscv_semihosting.c

```diff
@@ -110,32 +110,44 @@ enum semihosting_result riscv_semihosting(struct target *target, int *retval)
 	struct semihosting *semihosting = target->semihosting;
 	assert(semihosting);
 
-	if (!semihosting->is_active) {
-		LOG_TARGET_DEBUG(target, "Semihosting outcome: NONE (semihosting not enabled)");
-		return SEMIHOSTING_NONE;
-	}
-
 	riscv_reg_t pc;
 	int result = riscv_reg_get(target, &pc, GDB_REGNO_PC);
 	if (result != ERROR_OK) {
 		LOG_TARGET_DEBUG(target, "Semihosting outcome: ERROR (failed to read PC)");
 		return SEMIHOSTING_ERROR;
 	}
 
-	bool sequence_found;
+	bool sequence_found = false;
 	*retval = riscv_semihosting_detect_magic_sequence(target, pc, &sequence_found);
 	if (*retval != ERROR_OK) {
 		LOG_TARGET_DEBUG(target, "Semihosting outcome: ERROR (during magic seq. detection)");
 		return SEMIHOSTING_ERROR;
 	}
 
+	if (!semihosting->is_active) {
+		if (sequence_found) {
+			// If semihositing is encountered but disabled, provide an additional hint to the user.
```

Comment:
Typo: "semihositing".

Reply:
@TommyMurphyTM1234 this is already merged, so it's kind of late for review. No my suggestion to @JanMatCodasip is to consider reverting this patch if my arguments are considered good enough.